### PR TITLE
get-metadata: fail with a non-zero exit status

### PR DESCRIPTION
**Pull request by ryzokuken**
Fixes: https://github.com/nodejs/node-core-utils/issues/130

@joyeecheung @cPhost Please take a look. I hope this is what was needed to be done.

### CODE REVIEW

File: components/git/metadata.js [hunk 1]
```diff
@@ -83,15 +83,19 @@ function handler(argv) {
...
-    if (cli.spinner.enabled) {
-      cli.spinner.fail();
-    }
-    cli.error(err);
-    process.exit(-1);
-  });
+  return getMetadata(Object.assign({}, config, argv, parsed), cli)
+    .then(({status}) => {
+      if (status === false) {
+        process.exit(1);
```
Comment (joyeecheung) [inline components/git/metadata.js]: I think we can just let this throw here so it goes to the catch below, and replace `process.exit(-1)` below to `process.exit(-1)`? (I don't even know what I was thinking when I put `-1` there :/)

Comment (ryzokuken) [reply components/git/metadata.js]: I think you mean `process.exit(1)`?

Comment (joyeecheung) [reply components/git/metadata.js]: Yes, like so

```
    .then(({status}) => {
      if (status === false) {
        throw new Error(`PR checks failed`);
      }
    })
    .catch((err) => {
      if (cli.spinner.enabled) {
        cli.spinner.fail();
      }
      cli.error(err);
      process.exit(1);
    });

```

Comment (ryzokuken) [reply components/git/metadata.js]: Cool. That was exactly what I had been thinking about.

**Comment by ryzokuken** [comment]
@joyeecheung take a look now.

**Comment by BridgeAR** [comment]
@ryzokuken thanks a lot!

**Pull request merged**